<a href="https://colab.research.google.com/github/Luicazen/NLPTweets/blob/Rf/initLR.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [9]:
import numpy as np
import pandas as pd
from nltk.corpus import twitter_samples
import re
import string
from nltk.corpus import stopwords
from nltk.stem import PorterStemmer
from nltk.tokenize import TweetTokenizer
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.feature_extraction.text import CountVectorizer

In [2]:
from google.colab import files
uploaded = files.upload()
import pandas as pd
import numpy as np

Saving train.csv to train.csv


In [3]:
#Read in Training data
trainingDF = pd.read_csv('train.csv')
#Setting index to start at 1
trainingDF.index +=1
print("Training Data Summary")
#printing summary features: Shape
display(trainingDF.shape)
#Stats
display(trainingDF.describe())
#First 5 rows
display(trainingDF.head())
#type summary
trainingDF.info()
#column names
print(trainingDF.columns)

Training Data Summary


(7613, 5)

id      target
count   7613.000000  7613.00000
mean    5441.934848     0.42966
std     3137.116090     0.49506
min        1.000000     0.00000
25%     2734.000000     0.00000
50%     5408.000000     0.00000
75%     8146.000000     1.00000
max    10873.000000     1.00000

id keyword location                                               text  \
1   1     NaN      NaN  Our Deeds are the Reason of this #earthquake M...   
2   4     NaN      NaN             Forest fire near La Ronge Sask. Canada   
3   5     NaN      NaN  All residents asked to 'shelter in place' are ...   
4   6     NaN      NaN  13,000 people receive #wildfires evacuation or...   
5   7     NaN      NaN  Just got sent this photo from Ruby #Alaska as ...   

   target  
1       1  
2       1  
3       1  
4       1  
5       1

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7613 entries, 1 to 7613
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   id        7613 non-null   int64 
 1   keyword   7552 non-null   object
 2   location  5080 non-null   object
 3   text      7613 non-null   object
 4   target    7613 non-null   int64 
dtypes: int64(2), object(3)
memory usage: 297.5+ KB
Index(['id', 'keyword', 'location', 'text', 'target'], dtype='object')


In [4]:
#Check for null data
print("Training Data")
trainingDF.isnull().sum()


Training Data


id             0
keyword       61
location    2533
text           0
target         0
dtype: int64

In [10]:
#drop the target column from the training data X
X = trainingDF.drop(['target'], axis=1)
#Y is only the target column
y = trainingDF['target']

In [11]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)
X_train.shape, X_test.shape

((6090, 4), (1523, 4))

In [34]:

cv=CountVectorizer(min_df=1,max_df=1,ngram_range=(1,2))
#transformed train reviews
cv_train=cv.fit_transform(X_train['text'])
#transformed test reviews
cv_test=cv.transform(X_test['text'])

print('cv_train:',cv_train.shape)
print('cv_test:',cv_test.shape)

cv_train: (6090, 60905)
cv_test: (1523, 60905)


In [35]:
lr=LogisticRegression(penalty='l2',max_iter=500,C=1,random_state=0)
#Fitting the model for words
lr=lr.fit(cv_train,y_train)

In [37]:
from sklearn.metrics import accuracy_score, classification_report, f1_score
# Make predictions
lr_predict = lr.predict(cv_test)

# This model caluclates F1 score for both classifications. Compute weighted F1 score
lr_f1_score = f1_score(y_test, lr_predict, average='weighted')
print("F1 Score:", lr_f1_score)

F1 Score: 0.5937686735189955


Benchmark F1 score for an un-tuned Logistic Regression Model is: 0.5938